### ISSUE TITLE
[Q] Logging multiple arrays for each parameter in a sweep

### ISSUE THREAD

**Issue by sowmyamanojna** [wandb/client#1977]
I have been trying to log multiple arrays - loss (per epoch), accuracy (per epoch), validation loss (per epoch), validation accuracy (per epoch). When I log them separately like this:
```
for loss in nn_model.loss_hist:
    wandb.log({'loss': loss/t.shape[1]})

for acc in nn_model.accuracy_hist:
    wandb.log({'accuracy': acc/t.shape[1]})

for val_loss in nn_model.loss_hist_val:
    wandb.log({'val_loss': val_loss/t_val.shape[1]})

for val_accuracy in nn_model.accuracy_hist_val:
    wandb.log({'val_accuracy': val_accuracy/t_val.shape[1]})

for i in np.arange(config.num_epochs):
    wandb.log({"step":i})
```
`t_*` is the OneHotEncoded `y` matrix, used for normalization.

This is what I get:
![image](https://user-images.githubusercontent.com/31937434/111578683-a1acaa80-87da-11eb-9eb7-d39f237a5f63.png)

Notice the gap between the runs that have 5000 values in the arrays and the second set of runs that have 15000 values in the arrays.

When I try to log all of them in the same time, using zip:
```
for step_loss,step_acc,step_val_loss,step_val_accuracy,i in zip(nn_model.loss_hist,nn_model.accuracy_hist,nn_model.loss_hist_val,nn_model.accuracy_hist_val,np.arange(config.num_epochs)):
    wandb.log({'loss': step_loss/t.shape[1]})
    wandb.log({'accuracy': step_acc/t.shape[1]})
    wandb.log({'val_loss': step_val_loss/t_val.shape[1]})
    wandb.log({'val_accuracy': step_val_accuracy/t_val.shape[1]})
    wandb.log({"step":i})
```
This is what I get:
![image](https://user-images.githubusercontent.com/31937434/111577573-8c368100-87d8-11eb-8ccf-e4f060cee83a.png)

The problem being, the number of epochs is 10. So, the size of each array is only (10,1). However, from the graphs you can notice that the X-axis goes from 0-49 for the 5 arrays. The data is distributed in the following step fashion:

- loss: 1, 6, 11, ...
- accuracy: 2, 7, ...
and so on.

I tried changing the X-axis to `${config:step}` under the Expressions tab. But, with no avail.

Is there any workaround for this? 

Any help in this regard would be really helpful.

Thanks!

**Comment by vanpelt** [comment]
@sowmyamanojna if you want to plot metrics together on the same chart you need to log them to the same step.  Every time you call `wandb.log` we create a new step by default.  If you want to combine multiple calls to wandb.log into a single step you can pass `commit=False`, i.e.

```
    wandb.log({'loss': step_loss/t.shape[1]}, commit=False)
    wandb.log({'accuracy': step_acc/t.shape[1]}, commit=False)
    wandb.log({'val_loss': step_val_loss/t_val.shape[1]}, commit=False)
    wandb.log({'val_accuracy': step_val_accuracy/t_val.shape[1]}, commit=False)
    wandb.log({"step":i})
```

Then you can use "step" as the x-axis because those metrics are logged in the same logical wandb _step
